cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -91,8 +94,10 @@ protected void publish(Iterator<MetricValues> metrics) throws Exception {
       recordWriter.encode(metricValues, encoder);
       // Calculate the topic number with the hashcode of MetricValues' tags and store the encoded payload in the
       // corresponding list of the topic number
-      topicPayloads.get(Math.abs(metricValues.getTags().hashCode() % size))
-        .addPayload(encoderOutputStream.toByteArray());
+      int topicNumber = Math.abs(metricValues.getTags().hashCode() % size);
+      // this log is useful in knowing to which topic the program's metrics are emitted to
+      TOPIC_LOG.debug("Metrics is emitted to topic {}", topicPayloads.get(topicNumber).topicId.getTopic());
```

Comment:
do all metrics from the program go to the same topic? Otherwise I don't see how this is useful. But if that is the case, then we only need to compute it once at initialization time